### ISSUE TITLE
In-line/refreshable templates (data-view v1)

### ISSUE THREAD

**Issue by dwertheimer** [NotePlan/plugins#237]
someone could embed template code like this:
```Template
    some template code here
```
and then when NP sees that code block, it executes it (and refreshes it on demand). Whatever the return value is gets rendered by NotePlan in-line 

![Screen Shot 2022-03-08 at 1 06 47 PM](https://user-images.githubusercontent.com/8949588/157325662-772e15a7-5bee-492f-ab99-cc60cecda40a.jpg)

I was thinking that it would trigger a re-draw on:
- note load
- code block edit
- refresh button/menu press (initially could be on note `...` overflow menu, but at some point a circular refresh arrow on the block itself would be nice)

Note: @mikeerickson said: Another FYI, any time you execute a template to render, any JavaScript code you have between tags will be executed.

You will have access to any:

- NotePlan API command 
- Any methods defined in globals.js 
- Any methods you have supplied to in your own plugins located in methods object or variables located in data object 

So, basically using any template you can do d LOT LOT LOT LOT more than existing templates

**Comment by dwertheimer** [comment]
If it's gonna display the word "Template" maybe that's not the right word.
Maybe ```Data

**Comment by jgclark** [comment]
Instead I suggest this is a further use case for #187 (the first trigger).

**Comment by dwertheimer** [comment]
I am gonna let @mikeerickson respond to your comments, because he knows what Templates does and doesn't do. But in case I wasn't clear, I am talking about using np.Templates here inside of a code block. I suggested using \`\`\`Template or \`\`\`data to specifically delineate from standard JS code blocks, which by definition are just formatted display, not executed.
